plugins/device/rest.go

```diff
@@ -74,3 +74,65 @@ func deviceAddHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, peerInfo)
 }
+
+func groupEditHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	req := new(deviceapi.EditGroupReq)
+	if err := restutils.UnmarshalRequest(r, req); err != nil {
+		logger.WithError(err).Error("Failed to Unmarshal request")
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "Unable to marshal request", api.ErrCodeDefault)
+		return
+	}
+
+	peerID := mux.Vars(r)["peerid"]
+	if peerID == "" {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "peerid not present in request", api.ErrCodeDefault)
+		return
+	}
+	groupID := mux.Vars(r)["group_id"]
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+	lock, unlock, err := transaction.CreateLockSteps(peerID)
+	txn.Steps = []*transaction.Step{
+		lock,
+		{
+			DoFunc: "edit-group",
+			Nodes:  []uuid.UUID{gdctx.MyUUID},
+		},
+		unlock,
+	}
+	err = txn.Ctx.Set("peerid", peerID)
+	if err != nil {
+		logger.WithError(err).Error("Failed to set data for transaction")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+	err = txn.Ctx.Set("groupid", groupID)
+	if err != nil {
+		logger.WithError(err).Error("Failed to set data for transaction")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+	err = txn.Ctx.Set("req", req)
+	if err != nil {
+		logger.WithError(err).Error("Failed to set data for transaction")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+	err = txn.Do()
+	if err != nil {
+		logger.WithError(err).Error("Transaction to edit group failed")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "Transaction to edit group failed", api.ErrCodeDefault)
+		return
+	}
+	peerInfo, err := peer.GetPeer(peerID)
```

Comment:
PeerInfo response may not be necessory. Just return success with a message

Reply:
@rishubhjain 

Add peer and edit peer currently returns peerinfo. However, as this API is under devices, as suggested by Aravinda, it's okay to just return `http.StatusOK` without a body :+1: